sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1371 @@
+            .doOnSuccess(response -> logger.info("A batch of text sentiment output - {}", response))
+            .doOnError(error -> logger.warning("Failed to text sentiment - {}", error))
+            .map(response -> new SimpleResponse<>(response, toDocumentResultCollection(response.getValue())));
+    }
+
+    /**
+     * Get default country hint code.
+     *
+     * @return the default country hint code
+     */
+    public String getDefaultCountryHint() {
+        return defaultCountryHint;
+    }
+
+    /**
+     * Get default language when the builder is setup.
+     *
+     * @return the default language
+     */
+    public String getDefaultLanguage() {
+        return defaultLanguage;
+    }
+
+    private List<MultiLanguageInput> convertToMultiLanguageInput(List<TextDocumentInput> textInputs) {
+        List<MultiLanguageInput> multiLanguageInputs = new ArrayList<>();
+        for (TextDocumentInput textDocumentInput : textInputs) {
+            multiLanguageInputs.add(new MultiLanguageInput().setId(textDocumentInput.getId())
+                .setText(textDocumentInput.getText()).setLanguage(textDocumentInput.getLanguage()));
+        }
+        return multiLanguageInputs;
+    }
+
+    /**
+     * Helper method to convert the service response of {@link SentimentResponse} to {@link DocumentResultCollection}.
+     *
+     * @param sentimentResponse the {@link SentimentResponse} returned by the service.
+     *
+     * @return the {@link DocumentResultCollection} of {@link AnalyzeSentimentResult} to be returned by the SDK.
+     */
+    private DocumentResultCollection<AnalyzeSentimentResult> toDocumentResultCollection(
+        final SentimentResponse sentimentResponse) {
+        List<AnalyzeSentimentResult> analyzeSentimentResults = new ArrayList<>();
+        for (DocumentSentiment documentSentiment : sentimentResponse.getDocuments()) {
+            analyzeSentimentResults.add(convertToTextSentimentResult(documentSentiment));
+        }
+        for (DocumentError documentError : sentimentResponse.getErrors()) {
+            final com.azure.ai.textanalytics.models.TextAnalyticsError error = convertToError(documentError.getError());
+            analyzeSentimentResults.add(new AnalyzeSentimentResult(documentError.getId(), null, error, null,
+                null));
+        }
+        return new DocumentResultCollection<>(analyzeSentimentResults,
+            sentimentResponse.getModelVersion(), sentimentResponse.getStatistics() == null ? null
+            : mapBatchStatistics(sentimentResponse.getStatistics()));
+    }
+
+    private AnalyzeSentimentResult convertToTextSentimentResult(final DocumentSentiment documentSentiment) {
+        // Document text sentiment
+        final TextSentimentClass documentSentimentClass = TextSentimentClass.fromString(documentSentiment.
+            getSentiment().toString());
+        if (documentSentimentClass == null) {
```

Comment:
This would only happen if the service returned an invalid Sentiment type correct? Is this a good spot to throw a Runtime exception if the service failed? Maybe skip the result, log a warning, and continue processing the other responses.